packages/mobile/src/screens/chat-screen/ChatScreen.tsx

```diff
@@ -337,86 +339,101 @@ export const ChatScreen = () => {
             />
           ) : null}
         </Portal>
-        <View style={styles.rootContainer}>
-          {!isLoading ? (
-            chatMessages?.length > 0 ? (
-              <View style={styles.listContainer}>
-                <FlatList
-                  contentContainerStyle={styles.flatListContainer}
-                  data={chatMessages}
-                  keyExtractor={(message) => message.message_id}
-                  renderItem={({ item, index }) => (
-                    <>
-                      {/* When reaction popup opens, hide reaction here so it doesn't
+        <View
+          ref={chatContainerRef}
+          onLayout={() => {
+            chatContainerRef.current?.measureInWindow((x, y, width, height) => {
+              chatContainerTop.current = y
+            })
+          }}
+        >
+          <KeyboardAvoidingView
+            keyboardVerticalOffset={
+              Platform.OS === 'ios' ? chatContainerTop.current : 0
+            }
+            behavior={Platform.OS === 'ios' ? 'padding' : undefined}
```

Comment:
The docs recommend setting this explicitly for both iOS and Android (idk what to though)

Reply:
Noticed that but for android none of the options were working correctly, and I saw this is the pattern we use elsewhere.